lesson4/test_api/test_updating_resource_with_patch.py

```diff
@@ -0,0 +1,50 @@
+# Проверяем валидные id ресурсов
+def test_update_resource_with_patch_positive_id(session, base_url, fixture_id_resource_positive):
+    payload = {"title": "some title"}
+    res = session.patch(url=f'{base_url}/{fixture_id_resource_positive}', json=payload)
+    assert res.status_code == 200, "Status code is not 200"
+    assert res.json()["id"] == fixture_id_resource_positive, "Id is different"
+    assert res.json()["title"] == payload["title"], "Wrong title"
+
+
+# Проверяем замену значений всех ключей
+def test_update_resource_with_patch_positive_all_values(session, base_url):
+    payload = {"title": "NEW Title", "completed": False, "userId": 10}
+    res = session.patch(url=f'{base_url}/1', json=payload)
+    assert res.status_code == 200, "Status code is not 200"
+    assert res.json()["id"] == 1, "Id is not 1"
+    assert res.json()["title"] == payload["title"], "Wrong title"
+    assert res.json()["completed"] == payload["completed"], "Wrong completed status"
+    assert res.json()["userId"] == payload["userId"], "Wrong userId"
+
+
+# Проверяем замену части значений ключей, для остальных остаются старые значения
+def test_update_resource_with_patch_positive_few_values(session, base_url):
+    get_all_data = session.get(url=base_url)
+    payload = {"title": "NEW Title"}
+    res = session.patch(url=f'{base_url}/1', json=payload)
+    assert res.status_code == 200, "Status code is not 200"
+    assert res.json()["id"] == 1, "Id is not 1"
+    assert res.json()["title"] == payload["title"], "Wrong title"
+    assert res.json()["completed"] == get_all_data.json()[0]["completed"], "Wrong completed status"
+    assert res.json()["userId"] == get_all_data.json()[0]["userId"], "Wrong userId"
```

Comment:
You can save the first todo to the variable and use it for the assertion. That could be useful to improve code readability. For instance:
```
todo_1 = get_all_data.json()[0]
...
assert res.json()["completed"] == todo_1["completed"], "Wrong completed status"
assert res.json()["userId"] == todo_1["userId"], "Wrong userId"
```